src/components/GoalSidebar/GoalSidebar.tsx

```diff
@@ -88,86 +86,117 @@ export const GoalSidebar: FC<GoalSidebarProps> = ({ goal, onGoalTagRemove, onGoa
+                                renderTrigger={({ onClick }) =>
+                                    nullable(
+                                        safeUserData(goal.owner),
+                                        (props) => (
+                                            <UserBadge {...props}>
+                                                <IconXCircleSolid size="xs" onClick={onClick} />
+                                            </UserBadge>
+                                        ),
+                                        <AddInlineTrigger text={tr('Assign')} onClick={onClick} />,
+                                    )
+                                }
+                            />
+                        </StyledInlineInput>
+                    ),
+                    nullable(safeUserData(goal.owner), (props) => <UserBadge {...props} />, tr('Not assigned yet')),
                 )}
             </IssueMeta>
 
             {nullable(goal._isEditable || goal.participants.length, () => (
                 <IssueMeta title={tr('Participants')}>
-                    {goal.participants?.map((activity) =>
-                        nullable(safeUserData(activity), (props) => (
-                            <UserBadge key={activity.id} {...props}>
-                                {nullable(goal._isEditable, () => (
-                                    <IconXCircleSolid size="xs" onClick={onGoalParticipantRemove(activity.id)} />
-                                ))}
-                            </UserBadge>
-                        )),
-                    )}
+                    <TextList listStyle="none">
+                        {goal.participants?.map((activity) =>
+                            nullable(safeUserData(activity), (props) => (
+                                <TextListItem key={activity.id}>
+                                    <UserBadge {...props}>
+                                        {nullable(goal._isEditable, () => (
+                                            <IconXCircleSolid
+                                                size="xs"
+                                                onClick={onGoalParticipantRemove(activity.id)}
+                                            />
+                                        ))}
+                                    </UserBadge>
+                                </TextListItem>
+                            )),
+                        )}
+                    </TextList>
 
                     {nullable(goal._isEditable, () => (
                         <StyledInlineInput>
                             <UserComboBox
                                 placement="bottom-start"
                                 placeholder={tr('Type user name or email')}
                                 filter={participantsFilter}
-                                onChange={onGoalParticipantAdd}
+                                onChange={(activity) => onGoalParticipantAdd(activity)}
```

Comment:
isn't it better this way onChange={onGoalParticipantAdd} ?

Reply:
Strange things after rebase. Thanks 🙏🏼